ratatui-widgets/src/mascot.rs

```diff
@@ -0,0 +1,188 @@
+                MascotEye::Red => self.rat_eye_blink,
+            }),
+            '░' => Some(self.term_color),
+            '▓' => Some(self.term_cursor_color),
+            '▒' => Some(self.term_border_color),
+            _ => None,
+        }
+    }
+}
+
+impl Widget for RatatuiMascot {
+    /// Use half block characters to render a logo based on the `RATATUI_LOGO` const.
+    ///
+    /// The logo is rendered in three colors, one for the rat, one for the terminal, and one for the
+    /// rat's eye. The eye color alternates between two colors based on the selected row.
+    fn render(self, area: Rect, buf: &mut Buffer) {
+        for (y, (line1, line2)) in RATATUI_MASCOT.lines().tuples().enumerate() {
+            for (x, (ch1, ch2)) in line1.chars().zip(line2.chars()).enumerate() {
+                let x = area.left() + x as u16;
+                let y = area.top() + y as u16;
+                let cell = &mut buf[(x, y)];
+                // given two cells which make up the top and bottom of the character,
+                // Foreground color should be the non-space, non-terminal
+                let (fg, bg) = match (ch1, ch2) {
+                    (' ', ' ') => (None, None),
+                    (c, ' ') | (' ', c) => (self.color_for(c), None),
+                    (TERM, TERM_BORDER) => (self.color_for(TERM_BORDER), Some(Color::default())),
+                    (TERM, c) => (Some(Color::default()), self.color_for(c)), // treat the terminal background cells as empty
+                    (c, TERM) => (self.color_for(c), Some(Color::default())), // rather than having a character
+                    (c1, c2) => (self.color_for(c1), self.color_for(c2)),
+                };
+                // symbol should make the empty space or terminal bg as the empty part of the block
+                let symbol = match (ch1, ch2) {
+                    (TERM, TERM) => ' ',
+                    (_c, ' ' | TERM) => '▀',
+                    (' ' | TERM, _c) => '▄',
+                    (c, d) if c == d => '█',
+                    (_c, _d) => '▀',
+                };
+                if let Some(fg) = fg {
+                    cell.fg = fg;
+                }
+                if let Some(bg) = bg {
+                    cell.bg = bg;
+                }
+                cell.set_char(symbol);
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    #[test]
+    fn new_mascot() {
+        let mascot = RatatuiMascot::new();
+        assert_eq!(mascot.eye_state, MascotEye::Default);
+    }
+
+    #[test]
+    fn set_eye_color() {
+        let mascot = RatatuiMascot::new().set_eye(MascotEye::Red);
+        assert_eq!(mascot.eye_state, MascotEye::Red,);
+    }
+
+    #[test]
+    fn render_mascot() {
+        let mascot = RatatuiMascot::new();
+        let mut buf = Buffer::empty(Rect::new(0, 0, 32, 32));
+        mascot.render(buf.area, &mut buf);
+        assert_eq!(buf.area.as_size(), (32, 32).into());
+    }
+}
```

Comment:
I don't recall if we added a way to assert lines without style info in tests yet, but if there's something like that elsewhere in other tests, it would make a good test to check the output rendering is correct here.

Reply:
I added a check for the eye

Reply:
What I meant was
```rust
assert_eq!(buf, Buffer::with_lines([
    ".....",
    ".....",
]);
```

It's not a big deal to not have this though. Un-resolving so you see this comment. No action necessary.

Reply:
i've added a test to compare lines ;)